packages/flame/lib/src/timer.dart

```diff
@@ -1,8 +1,6 @@
 import 'dart:math';
 import 'dart:ui';
 
-import 'components/component.dart';
-
 /// Simple utility class that helps handling time counting and implementing
 /// interval like events.
```

Comment:
The purpose is unclear: how does this help with interval-like events? Or is this class mixes two functionalities: keeping track of time and being an EffectController?

Reply:
You use the `callback` to help with "interval like events", it is more general than the effect controller. Say that you want to spawn some new components every 5 seconds, then this is useful.

Reply:
It still sounds like an effect, though. Call it `CallbackEffect`, and let it invoke the provided callback function when the progress reaches 1. The code would be very simple and similar to `RemoveEffect`.

But unlike the current `TimerComponent`, it will allow you to customize the timing more flexibly. For example, you can make the callback fire once, or `n` times, or keep firing infinitely; the intervals between invocations can be either the same, or random (once we have `RandomEffectController`), or custom (if you define your own controller).

Reply:
TimerComponent is just a wrapper on the Timer class which is a simpler, generic feature for running period things, it is simpler and it is available outside the FCS too.

Both things can exist, if what you need is too complex for the TimerComponent to be used, effects can be used instead.